pkg/internalregistry/controllers/keyid_observation_controller.go

```diff
@@ -104,6 +110,38 @@ func (c *keyIDObservation) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+func (c *keyIDObservation) fallbackKeyIDObservation(ctx context.Context) error {
+	_, err := c.secrets.Secrets("openshift-kube-apiserver").Get("bound-service-account-signing-key")
+	if err != nil && !errors.IsNotFound(err) {
+		// signing key secret exists, skip
+		return err
+	}
```

Comment:
Is it safe to make any assumptions one way or the other about the secret's existence based on this information?

Reply:
We shouldn't reach here unless the cache has synced. If the secret should exist, but doesn't at this point, I don't see the harm in performing the fallback here, and the regular control loop will eventually pick up the secret and override if needed.

Reply:
I think there are 3 cases:
1. The secret exists (`err == nil`) and the fallback is performed. The regular control loop should eventually catch up.
2. The secret does not exist (`errors.IsNotFound(err)`) and the fallback is performed. Even if the secret is later created, the regular control loop takes over.
3. Some different error occurred, and the fallback is _not_ performed. This seems like a problem on a cluster that will never have the signing key secret.

Reply:
I see it now. Fixed.